src/app/modules/main/wallet_section/transactions/view.nim

```diff
@@ -66,18 +67,24 @@ QtObject:
     return self.model.getCount() > 0
 
   proc loadTransactionsForAccount*(self: View, address: string, toBlock: string = "0x0", limit: int = 20, loadMore: bool = false) {.slot.} =
-    self.setHistoryFetchState(address, true)
+    self.setHistoryFetchState(address, allTxLoaded=not self.models[address].getHasMore(), isFetching=true)
+    self.models[address].addPageSizeBuffer(limit)
     self.delegate.loadTransactions(address, toBlock, limit, loadMore)
 
   proc setTrxHistoryResult*(self: View, transactions: seq[Item], address: string, wasFetchMore: bool) =
     if not self.models.hasKey(address):
       self.models[address] = newModel()
 
     self.models[address].addNewTransactions(transactions, wasFetchMore)
+    if self.fetchingHistoryState[address] and wasFetchMore:
```

Comment:
@Khushboo-dev-cpp there are few places here where we're accessing the value by key (for models and fetchingHistoryState) without checking if the key is present in the table, are we sure there is no case where in the moment of checking the searched key is not in the table?

Reply:
I will double check this, I had tried to add wherever possible it will crash otherwise, thanks for the comment